tests/components/mikrotik/test_config_flow.py

```diff
@@ -0,0 +1,192 @@
+"""Test Mikrotik setup process."""
+from unittest.mock import patch
+
+import librouteros
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components import mikrotik
+from homeassistant.components.mikrotik import config_flow
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+
+from tests.common import MockConfigEntry
+
+DEMO_USER_INPUT = {
+    CONF_NAME: "Home router",
+    CONF_HOST: "0.0.0.0",
+    CONF_USERNAME: "username",
+    CONF_PASSWORD: "password",
+    CONF_PORT: 8278,
+    CONF_VERIFY_SSL: False,
+    mikrotik.CONF_TRACK_DEVICES: True,
+}
+
+DEMO_CONFIG = {
+    CONF_NAME: "Home router",
+    CONF_HOST: "0.0.0.0",
+    CONF_USERNAME: "username",
+    CONF_PASSWORD: "password",
+    CONF_PORT: 8278,
+    CONF_VERIFY_SSL: False,
+    mikrotik.const.CONF_FORCE_DHCP: False,
+    mikrotik.CONF_TRACK_DEVICES: True,
+    mikrotik.CONF_ARP_PING: False,
+    mikrotik.CONF_DETECTION_TIME: 30,
+}
+
+
+MOCK_ENTRY = MockConfigEntry(domain=mikrotik.DOMAIN, data=DEMO_CONFIG)
+
+
+@pytest.fixture(name="api")
+def mock_mikrotik_api():
+    """Mock an api."""
+    with patch("librouteros.connect"):
+        yield
+
+
+@pytest.fixture(name="auth_error")
+def mock_api_authentication_error():
+    """Mock an api."""
+    with patch(
+        "librouteros.connect",
+        side_effect=librouteros.exceptions.TrapError("invalid user name or password"),
+    ):
+        yield
+
+
+@pytest.fixture(name="conn_error")
+def mock_api_connection_error():
+    """Mock an api."""
+    with patch("transmissionrpc.Client", side_effect=librouteros.exceptions.TrapError):
+        yield
+
+
+def init_config_flow(hass):
+    """Init a configuration flow."""
+    flow = config_flow.MikrotikFlowHandler()
+    flow.hass = hass
+    return flow
+
+
+async def test_import(hass, api):
+    """Test import step."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_import(DEMO_CONFIG)
```

Comment:
Please use the core interfaces, in this case `hass.config_entries.flow.async_init`, to set up the tests.

```py
result = await hass.config_entries.flow.async_init(DOMAIN, context={"source": "import"}, data=DEMO_CONFIG)
```

https://github.com/home-assistant/home-assistant/blob/1fffa210e1617b516d8ac10fe2c59e8838391935/homeassistant/data_entry_flow.py#L94-L96